docs-sphinx/basic.rst

```diff
@@ -1251,3 +1251,92 @@ In addition, each TBranch of the TTree can have a different compression setting:
     {'x': None, 'ny': None, 'y': ZLIB(4)}
 
 Changes to the compression setting only affect TBaskets written after the change (with :ref:`uproot.writing.writable.WritableTree.extend`; see above).
+
+Using fsspec for reading and writing files
+--------------------------
+
+Since version `5.2.0 <https://github.com/scikit-hep/uproot5/releases/tag/v5.2.0>`_, uproot supports reading and writing files using `fsspec <https://filesystem-spec.readthedocs.io/en/latest/>`_.
+This allows you to read and write files from a variety of sources, including cloud storage, HTTP, and more.
+
+Usage of fsspec as a source is the default behaviour since 5.2.0, but the user is able to manually specify the source by passing a `uproot.source.chunk.Source` class to the `handler` argument of different uproot methods, such as `uproot.open`, `uproot.iterate`, `uproot.concatenate`, etc.
+
+In general the user should not need to worry about the source, as uproot will automatically choose the best source for the given path.
+
+In some cases it may provide a performance benefit to manually specify the source, for example when opening a file from a local path, specifying `handler=uproot.source.file.MemmapSource` (instead of the default `handler=uproot.source.fsspec.FSSpecSource`) may reduce the time to open the file at the cost of using more memory.
+
+Any fsspec protocol should work for reading, while only the protocols supporting writing will work for writing.
+
+fsspec is a dependency of uproot, but in order to use some protocols, the user may need to install additional dependencies.
+For example, in order to open S3 files, the user needs to have `s3fs <https://github.com/fsspec/s3fs>`_ installed.
+When attempting to open a file with a protocol that is not supported, uproot will raise an exception with a helpful message pointing towards the missing dependency.
+
+For some protocols, such as `s3` or `ssh`, fsspec may need additional options, such as credentials. These can be directly passed as keyword arguments to the uproot function, and will be passed to fsspec.
```

Comment:
This is good information, though it should come after a basic "how to" sentence. The first thing readers of this documentation should see is a statement saying that they can use any remote protocol that fsspec ([link](https://filesystem-spec.readthedocs.io/)) knows about by writing a URL schema, such as "`https://`", "`root://`", or "`s3://`".

Second, you'd say that they might be prompted to load additional dependencies.

Third, you'd say that some of these protocols will work for writing; see the fsspec documentation for details.

Fourth, that some of them require additional arguments, and they can be passed as keyword `**options`.

Reply:
The comment on writing could also indicate that this is possible, but likely inefficient. The best way to write a file and host it remotely is to write it locally and upload the final result.